### DISCUSSION TITLE
Are nested events possible?

### DISCUSSION THREAD

**Discussion by desiwko** [thepirat000/Audit.NET#582] [Q&A]
I haven't been able to find many comprehensive examples of Audit.NET to guide me though what's in the readme is plenty enough to get started.  What I'm looking for in particular but not able to find is whether there is a way to create nested single-file audit flows, or otherwise definitively link audits together in some sane way.

I'm looking at a concurrent system where multiple operations will be in flight at once from different sources.  It would be great to find a way to organize those audit trails into something more meaningful than dumping each action to an individual json file somewhere.  Ideally starting at a parent and then nesting audits within it would be nice.  I'd be interested if this is supported or if there are any other possible solutions out of the box.

**Comment by thepirat000** [comment]
Your question seems too broad. Have you read the docs, especially the [Data Providers](https://github.com/thepirat000/Audit.NET#data-providers) section?

Saving the audits to JSON files is just the default behavior, but you can choose from [more than 20 data providers](https://github.com/thepirat000/Audit.NET#storage-providers) or create your own.

Regarding the audit nesting, I'm not sure what problem you are solving, but generally speaking, the way to correlate audit logs is [using a correlation id](https://www.oreilly.com/library/view/building-microservices-with/9781785887833/1bebcf55-05bb-44a1-a4e5-f9733b8edfe3.xhtml#:~:text=A%20correlation%20ID%20is%20a,services%20using%20the%20logging%20information.).
